tasks/configuration.yml

```diff
@@ -24,16 +24,18 @@
 
 - name: add our own host configuration
   template:
-    src: templates/etc/icinga2/conf.d/hosts.conf.j2
-    dest: /etc/icinga2/conf.d/hosts.conf
+    src: templates/etc/icinga2/zones.d/generic_host.conf.j2
+    dest: '/etc/icinga2/zones.d/{{ hostvars[item].icinga2_client_parent_zone }}/{{ hostvars[item].inventory_hostname }}_host.conf'
```

Comment:
What kind of variable is `icinga2_client_parent_zone`? Cross-referencing variables of another role is a bad idea if they're not explicitly defined as a dependency in the meta main.yml.